test/dns-test.js

```diff
@@ -0,0 +1,396 @@
+
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  rsNoUnbound: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+const nstub = new StubResolver({
+  rd: true,
+  cd: true,
+  edns: true,
+  ednsSize: 4096,
+  maxAttempts: 2,
+  maxTimeout: 3000,
+  dnssec: true,
+  servers: [`127.0.0.1:${network.nsPort}`]
+});
+
+const rstub = new StubResolver({
+  rd: true,
+  cd: true,
+  edns: true,
+  ednsSize: 4096,
+  maxAttempts: 2,
+  maxTimeout: 3000,
+  dnssec: true,
+  servers: [`127.0.0.1:${network.rsPort}`]
+});
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const wallet = wclient.wallet('primary');
+
+const {
+  treeInterval,
+  biddingPeriod,
+  revealPeriod
+} = network.names;
+
+// miner controlled address
+let coinbase;
+
+describe('DNS Servers', function() {
+  this.timeout(15000);
+
+  before(async () => {
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+    await nstub.open();
+    await rstub.open();
+
+    // mine a bunch of blocks
+    const info = await wallet.createAddress('default');
+    coinbase = info.address;
+
+    for (let i = 0; i < 15; i++)
+      await nclient.execute('generatetoaddress', [2, coinbase]);
+  });
+
+  after(async () => {
+    await nclient.close();
+    await wclient.close();
+    await node.close();
+    await nstub.close();
+    await rstub.close();
+  });
+
+  describe('Recursive Resolver', () => {
+    it('should return an A record', async () => {
+      const name = 'google.com.';
+
+      const res = await rstub.lookup(name, types.A);
+
+      const [question] = res.question;
+      assert(typeof question === 'object');
+      assert.equal(question.name, name);
+
+      const [answer] = res.answer;
+      assert(typeof answer === 'object');
+      assert.equal(answer.name, name);
+    });
+  });
+
+  describe('Authoritative Resolver', () => {
+    it('should return no answer for an ICANN rooted name', async () => {
+      const name = 'com.';
+
+      const res = await nstub.lookup(name, types.A);
+
+      assert.equal(res.question[0].name, name);
+      assert.equal(res.answer.length, 0);
+
+      // Handshake zone is not authoritative for ICANN based name
+      assert.equal(res.aa, false);
+    });
+
+    it('should return no answer for HNS name pre-update', async () => {
+      const name = 'foo.';
+
+      const res = await nstub.lookup(name, types.A);
+      assert.equal(res.answer.length, 0);
+    });
+
+    // record to save in the authenticated tree
+    for (const [hstype, item] of Object.entries(json)) {
+      const type = types[item.type];
+      const resource = Resource.fromJSON(item.resource);
+
+      // args: size, height, network
+      const name = rules.grindName(5, 1, network);
+
+      it(`should update ${name}`, async () => {
```

Comment:
I don't think the `${name}` is useful here since it's randomly generated, and each pass of the test will be different.
Perhaps "should update name with record type ${item.type}"

Reply:
These are all written directly to the tree now in a `before` block, so this is on longer relevant.